packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/FileSystemGraphObjectStore.ts

```diff
@@ -160,13 +224,29 @@ export class FileSystemGraphObjectStore implements GraphObjectStore {
       pMap(
         this.localGraphObjectStore.collectEntitiesByStep(),
         async ([stepId, entities]) => {
-          await flushDataToDisk({
-            storageDirectoryPath: stepId,
-            collectionType: 'entities',
-            data: entities,
-            pretty: this.prettifyFiles,
+          const indexable = entities.filter((e) => {
+            const indexMetadata = this.getIndexMetadataForGraphObjectType({
+              stepId,
+              _type: e._type,
+              graphObjectCollectionType: 'entities',
+            });
+
+            if (typeof indexMetadata === 'undefined') {
+              return true;
+            }
+
+            return indexMetadata.enabled === true;
           });
 
+          if (indexable.length) {
```

Comment:
Do you think it would be worthwhile to add a test for the scenario when there are no indexable types?

Reply:
Will do!